website/content/docs/nia/configuration.mdx

```diff
@@ -330,6 +330,14 @@ driver "terraform-cloud" {
   - We recommend creating a dedicated team and team API token to isolate automation by Consul-Terraform-Sync from other Terraform Cloud operations.
 - `workspace_prefix` - (string) Specifies a prefix to prepend to the automatically-generated workspace names used for automation. This prefix will be used by all tasks that use this driver. By default, when no prefix is configured, the workspace name will be the task name. When a prefix is configured, the workspace name will be `<workspace_prefix value><task name>`. For example, if you configure the prefix as "cts_", then a task with the name "task_firewall" will have the workspace name "cts_task_firewall".
 - `required_providers` - (obj: required) Declare each Terraform provider used across all tasks. This can be configured the same as how you would configure [Terraform `terraform.required_providers`](https://www.terraform.io/docs/configuration/provider-requirements.html#requiring-providers) field to specify the source and version for each provider. Consul-Terraform-Sync will process these requirements when preparing each task that uses the provider.
+- `tls` - Configure TLS to allow HTTPS connections to [Terraform Enterprise](https://www.terraform.io/docs/enterprise/install/installer.html#tls-key-amp-cert).
+  - `enabled` - (bool) Enable TLS. Specifying any option for TLS will also enable it.
+  - `verify` - (bool: true) Enables TLS peer verification. The default is enabled, which will check the global CA chain to make sure the given certificates are valid. If you are using a self-signed certificate that you have not added to the CA chain, you may want to disable SSL verification. However, please understand this is a potential security vulnerability.
```

Comment:
```suggestion
  - `verify` - (bool: true) Enables TLS peer verification. The default is enabled, which will check the global CA chain to make sure the given certificates are valid. If you are using a self-signed certificate that you have not added to the CA chain, you may want to disable SSL verification. However, please understand this is a potential security vulnerability.
```

Is the reasoning for wanting to disable because an error will be returned? What's the expected behavior?

Reply:
When using self-signed certificates, it results in an error like this:
```
2021/09/13 16:32:02.588083 [ERR] (driver.tfc) error creating Terraform Cloud client for verification: Get "https://tfe.local/api/v2/ping": x509: certificate relies on legacy Common Name field, use SANs or temporarily enable Common Name matching with GODEBUG=x509ignoreCN=0
```

```hcl
  tls {
    verify = false
  }
```

Setting verify to false allows CTS to reach out to TFE and ignore any certificate warnings.

Reply:
> When using self-signed certificates, it results in an error like this:
> 
> ```
> 2021/09/13 16:32:02.588083 [ERR] (driver.tfc) error creating Terraform Cloud client for verification: Get "https://tfe.local/api/v2/ping": x509: certificate relies on legacy Common Name field, use SANs or temporarily enable Common Name matching with GODEBUG=x509ignoreCN=0
> ```
> 
> ```terraform
>   tls {
>     verify = false
>   }
> ```
> 
> Setting verify to false allows CTS to reach out to TFE and ignore any certificate warnings.

We should call that out then so it's clearer for the practitioner.

Reply:
Sounds good, added the reason. I split up the self-signed cert portion too

```suggestion
  - `verify` - (bool: true) Enables TLS peer verification. The default is enabled, which will check the global CA chain to make sure the given certificates are valid.
    - If you are using a self-signed certificate that you have not added to the CA chain, you may want to disable SSL verification to ignore any certificate warnings. However, please understand this is a potential security vulnerability.
      ```hcl
      tls {
        verify = false
      }
      ```
```